### PR TITLE
Fix #3557: Merge profile chooser fragment into 2 fragments

### LINKED ISSUE #3557: Merge profile_chooser_fragment.xml into single xml file
Currently there are 4 versions of `profile_chooser_fragment.xml` file merge it into single xml file.

We can use https://text-compare.com/ to compare two versions of this file and for all the differences we can create variables in `dimens.xml` file and use it accordingly.

It can happen that it becomes really hard to merge all 4 files into single file, in such cases prefer code-readability over single file. Therefore it can be fine if in the end the code still contains 2 or 3 files.

**Note**: **In PR, make sure you add before and after screenshot of mobile-portrait, mobile-landscape, tablet-portrait and tablet-landscape for comparison and make sure that there is not difference between before and after UI.**

### PR DESCRIPTION
**Pull request by priyatanu**
## Explanation
Merged 2 files of profile_chooser_fragment.xml into one profile_chooser_fragment.xml in layout/profile_chooser_fragment.xml
But for the larger screen size I found that sw600dp profile_chooser_fragment is different to mobile screen layout. So I have merged 2 sw600dp modes profile_chooser_fragment into 1 profile_chooser_fragment.xml inside sw600dp folder

## Essential Checklist

- [ ] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [ ] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [ ] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [ ] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [ ] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by priyatanu** [comment]
@rishidyno @ARJUPTA

**Comment by priyatanu** [comment]
Screenshot_2
Screenshot_3
Screenshot1

**Comment by priyatanu** [comment]
Screenshot_4

**Comment by priyatanu** [comment]
TestCase1

**Comment by priyatanu** [comment]
Hi @ARJUPTA @rishidyno ,

kindly review my pr

**Comment by priyatanu** [comment]
Hi @BenHenning , kindly review my pr

**Comment by BenHenning** [comment]
FYI @priyatanu you should include 'PTAL' in your message when you'd like someone to take a look at the PR (since then Oppiabot will automatically assign it).

File: app/src/main/res/values-land/styles.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+  <style name="Heading1" parent="TextView.Common1">
```
Comment (BenHenning) [inline app/src/main/res/values-land/styles.xml:3]: I'm not sure that we actually want to change this for all landscape headings. Why is this needed?

FWIW I think we can just use 'Heading1ViewStart' since that's the one we should always be using for non-centered headings.

File: app/src/main/res/values/dimens.xml [hunk 2]
```diff
@@ -459,6 +459,18 @@
   <dimen name="profile_chooser_fragment_profile_select_text_margin_top">64dp</dimen>
   <dimen name="profile_chooser_fragment_profile_recycler_view_margin_top">36dp</dimen>
   <dimen name="profile_chooser_profile_view_view_margin_top">60dp</dimen>
+  <dimen name="profile_chooser_profile_recycler_view_margin_start">32dp</dimen>
+  <dimen name="profile_chooser_profile_recycler_view_margin_end">32dp</dimen>
+  <dimen name="profile_chooser_profile_view_layout_height">128dp</dimen>
+  <dimen name="profile_chooser_profile_select_text_margin_start">64dp</dimen>
+  <dimen name="profile_chooser_fragment_text_view_layout_margin_bottom">32dp</dimen>
+  <dimen name="profile_chooser_fragment_image_view_layout_margin_end">64dp</dimen>
```
Comment (BenHenning) [inline app/src/main/res/values/dimens.xml:467]: Are these views not also present in the non-tablet versions? It'd be nice to consolidate to as few constants as possible.

File: app/src/main/res/layout-sw600dp-land/profile_chooser_fragment.xml [hunk 3]
```diff
@@ -1,125 +0,0 @@
...
-          android:orientation="vertical"
-          android:overScrollMode="never"
-          android:paddingBottom="@{hasProfileEverBeenAddedValue ? @dimen/profile_chooser_padding_bottom_profile_already_added: @dimen/bottom_white_space}"
-          android:requiresFadingEdge="vertical"
-          android:scrollbars="none"
-          android:tag="profile_recycler_view"
-          app:data="@{viewModel.profiles}"
-          app:layoutMarginEnd="@{hasProfileEverBeenAddedValue ? @dimen/profile_chooser_margin_end_profile_already_added : @dimen/space_0dp}"
-          app:layoutMarginStart="@{hasProfileEverBeenAddedValue ? @dimen/profile_chooser_margin_start_profile_already_added : @dimen/space_0dp}"
-          app:layoutMarginTop="@{hasProfileEverBeenAddedValue ? @dimen/space_0dp : @dimen/profile_chooser_margin_top_profile_not_added}"
```
Comment (BenHenning) [inline app/src/main/res/layout-sw600dp-land/profile_chooser_fragment.xml:84]: This part is inconsistent with the tablet portrait version--it seems like it will be a regression per the diff. Is the new result correct?

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 4]
```diff
@@ -25,6 +25,17 @@
...
+
+  <!-- Profile Chooser Fragment -->
+  <dimen name="profile_chooser_fragment_layout_margin_start">96dp</dimen>
+  <dimen name="profile_chooser_fragment_layout_margin_bottom">28dp</dimen>
+  <dimen name="profile_chooser_fragment_layout_margin_end">96dp</dimen>
+  <dimen name="profile_chooser_profile_select_text_margin_start">96dp</dimen>
+  <dimen name="profile_chooser_fragment_margin_top_profile_already_added">24dp</dimen>
+  <dimen name="profile_chooser_fragment_margin_top_profile_not_added">84dp</dimen>
+  <dimen name="profile_chooser_fragment_text_view_layout_margin_bottom">28dp</dimen>
+  <dimen name="profile_chooser_fragment_image_view_layout_margin_end">96dp</dimen>
```
Comment (BenHenning) [inline app/src/main/res/values-sw600dp-land/dimens.xml:37]: Are all of these being used in the tablet views?

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @priyatanu. I took a first pass and left some comments. Can you please also add screenshots to your main PR description under the UI section with before-and-after screenshots (i.e. screenshots of the app before and after the changes so that we can see the difference)? Please also fill out the checkboxes and make sure all aspects of the PR description have been filled in as instructed.

Regarding not being able to merge the tablet & handset versions of the layout--is that actually true? After looking at a text comparison of the two, the main differences seem to be:
- An extra variable for whether a profile was ever opened (adding this in the handset version of the layout should be fine if it isn't used or is properly defaulted)
- Some style & sizing differences (which can be split based on density-qualified resources)
- Some ConstraintLayout constraints (similarly, I think these can be split using qualifier values files as well, or perhaps we might need to set them based on a databinding value)
- Some margin differences based on the profile boolean--I wonder if these can use dimensions that just default to whatever they would need to in order to match current portrait/landscape dimensions

It seems possible to reduce all 4 layouts to 1, and that would be a really nice simplification.

**Comment by priyatanu** [comment]
> Thanks @priyatanu. I took a first pass and left some comments. Can you please also add screenshots to your main PR description under the UI section with before-and-after screenshots (i.e. screenshots of the app before and after the changes so that we can see the difference)? Please also fill out the checkboxes and make sure all aspects of the PR description have been filled in as instructed.
> 
> Regarding not being able to merge the tablet & handset versions of the layout--is that actually true? After looking at a text comparison of the two, the main differences seem to be:
> 
> * An extra variable for whether a profile was ever opened (adding this in the handset version of the layout should be fine if it isn't used or is properly defaulted)
> * Some style & sizing differences (which can be split based on density-qualified resources)
> * Some ConstraintLayout constraints (similarly, I think these can be split using qualifier values files as well, or perhaps we might need to set them based on a databinding value)
> * Some margin differences based on the profile boolean--I wonder if these can use dimensions that just default to whatever they would need to in order to match current portrait/landscape dimensions
> 
> It seems possible to reduce all 4 layouts to 1, and that would be a really nice simplification.

Thank you, I will look into this
